### DISCUSSION TITLE
Automatically gzip Wasm binaries

### DISCUSSION THREAD

**Discussion by lastmjs** [dfinity/sdk#3110] [Feature Requests]
dfx 0.14.0 introduced automatic binary optimization from wasm-opt with a new `optimize` field in the dfx.json. This is great, but the wasm-opt might not be enough. gzip may be necessary after the wasm-opt pass, but it is difficult to retrieve the binary after optimization as it's stored rather opaquely in `.dfx` (actually is that true? I think it might be optimizing the binary in-place using the path to the binary from dfx.json).

It would be great if gzip could be done automatically in addition to wasm-opt, to have even more optimization tools available easily to developers.

**Comment by adamspofford-dfinity** [comment]
This would have to be an explicit field, as gzipping the canister is meaningful on the replica's end - the reported canister module hash is that of the gz file, not of the uncompressed wasm file.

**Reply by lastmjs** [reply]
Sounds good to me, just like we have the `optimize` field, we could have a `gzip` field. If they are both specified, optimization would happen first and then gzip. And gzip perhaps should allow different settings just like `optimize`, i.e. `-9`.

**Comment by kentosugama** [comment]
Just to clarify, the binary is optimized in place

**Comment by ghost** [comment]
`gzip: true` canister field added in dfx 0.14.1
